File: currency_parser.rb
#!/home/mariusz/.rbenv/shims/ruby

require 'rubygems'
require 'nokogiri'
require 'restclient'

class CurrencyParser
   def initialize(currency)
      @page=Nokogiri::HTML(RestClient.get("http://kantor.aliorbank.pl/forex"))
      @currency=currency
      @sell_list=@page.css('td.ere-sale')
      @buy_list=@page.css('td.ere-purchase')
   end

   def display_sell()
      puts "SELL\t1 " + @currency + " = " + get_price(@sell_list) + " PLN"
   end

   def get_price(list)
      return list.css('span.__currency-rate').css('span[data-currency=' + @currency + ']').text
   end

   def display_buy()
      puts "BUY\t1 " + @currency + " = " + get_price(@buy_list) + " PLN"
   end
   
end

if ARGV.empty?
   puts "You must pass an argument for currency ie. \"GBP\""
   exit
else 
   CURRENCY = ARGV[0]
end

if ARGV[0].length != 3
   puts "Currency code should contain only 3 characters ie. \"GBP\""
   exit
end

parser=CurrencyParser.new(CURRENCY)
parser.display_sell()
parser.display_buy()

File: currency_parser.py
#!/usr/bin/python

import sys
from bs4 import BeautifulSoup
from six.moves import urllib

class CurrencyParser:
   
   def __init__(self, currency):
      self.currency = currency
      self.page = urllib.request.urlopen('http://kantor.aliorbank.pl/forex')
      self.soup = BeautifulSoup(self.page, "lxml")
      self.sellList = self.soup.body.findAll('td', attrs={'class':'ere-sale'})
      self.buyList = self.soup.body.findAll('td', attrs={'class':'ere-purchase'})

   def getCurrencyFromList(self, searchList):
      for x in searchList:
         curr = x.find('span', attrs={'data-currency':self.currency})
         if curr is not None:
            return curr.text

   def getPrice(self, sourceList):
      value = self.getCurrencyFromList(sourceList)
      if value is None:
         raise "Invalid currency value!"
      return value

   def displaySell(self):
      print "SELL\t1 " + self.currency + " = " + self.getPrice(self.sellList) + " PLN"

   def displayBuy(self):
      print "BUY\t1 " + self.currency + " = " + self.getPrice(self.buyList) + " PLN"


if len(sys.argv) == 2:
   currencyArgument = sys.argv[1]
else:
   print "You must pass an argument for currency ie. \"GBP\"" 
   exit()

if len(currencyArgument) != 3:
   print "Currency code should contain only 3 characters ie. \"GBP\""
   exit()

parser = CurrencyParser(currencyArgument)
parser.displaySell()
parser.displayBuy()
